float64 tx
float64 ty
float64 tz
---
bool success